internal/config/dns/webhook_dns.go

```diff
@@ -142,83 +155,160 @@ func (c *OperatorDNS) Delete(bucket string) error {
 	}
 	xhttp.DrainBody(resp.Body)
 	if resp.StatusCode != http.StatusOK {
+		switch resp.StatusCode {
+		case http.StatusNotFound:
+			return ErrNoEntriesFound
+		}
 		return fmt.Errorf("request to delete the service for bucket %s, failed with status %s", bucket, resp.Status)
 	}
 	return nil
 }
 
-// DeleteRecord - Removes a specific DNS entry
-// No Op for Operator because operator deals on with bucket entries
-func (c *OperatorDNS) DeleteRecord(record SrvRecord) error {
-	return ErrNotImplemented
-}
-
 // Close closes the internal http client
-func (c *OperatorDNS) Close() error {
+func (c *WebhookDNS) Close() error {
 	c.httpClient.CloseIdleConnections()
 	return nil
 }
 
 // List - Retrieves list of DNS entries for the domain.
-// This is a No Op for Operator because, there is no intent to enforce global
-// namespace at MinIO level with this DNS entry. The global namespace in
-// enforced by the Kubernetes Operator
-func (c *OperatorDNS) List() (srvRecords map[string][]SrvRecord, err error) {
-	return nil, ErrNotImplemented
+func (c *WebhookDNS) List() (srvRecords map[string][]SrvRecord, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultWebhookContextTimeout)
+	defer cancel()
+	ep, err := c.endpoint("", false)
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, ep, nil)
+	if err != nil {
+		return nil, err
+	}
+	if err = c.addAuthHeader(req); err != nil {
+		return nil, err
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer xhttp.DrainBody(resp.Body)
+
+	if resp.StatusCode != http.StatusOK {
+		// This is needed because our operator webhook
+		// does not implement these methods.
+		return nil, ErrNoEntriesFound
```

Comment:
Not sure - it looks like we return ErrNoEntriesFound for any status code here

Reply:
anything other than 200 OK - its a protocol

Reply:
Also, this function is not necessary, only the ones that implement it.